app/models/game.rb

```diff
@@ -23,31 +22,11 @@ def populate_board!
     end
   end
 
-  def check?(color)
-    if color == 'white'
-      king_x = @white_king.horizontal_position
-      king_y = @white_king.vertical_position
-    else
-      king_x = @black_king.horizontal_position
-      king_y = @black_king.vertical_position
-    end
-    Piece.where(!color).each do |piece|
-      return true if piece.valid_move?(king_x, king_y)
+  def check?
+    king = King.where(game: self, color: color)
```

Comment:
I think "where" will return an array, so we need to get just one result:
king = King.where(game: self, color: color).first